### ISSUE TITLE
Suggestions for OSSU improvement

### ISSUE THREAD

**Issue by ericdouglas** [ossu/roadmap#5]
Continuing the discussion started [here](https://github.com/ossu/computer-science/issues/418).

**Comment by scamargo** [comment]
A lot of people from the previous post mentioned a need for community. Do we have a public slack/gitter group?

**Comment by ericdouglas** [comment]
Yes, we do. You can see all channels [here](https://github.com/ossu/computer-science#community).

**Comment by bewwys** [comment]
Hello @ericdouglas ,
Thank you so much for all staff that you put here :) and sorry for my bad english.
So, what I'm feeling right now it's the lack of projects. Yeah I can go to codewars, topcoder and so on... but I really feel like I'm not doing what an employer want me to do. I think we need labs (progressive labs). We need to be in reals conditions.

It's difficult to explain what I'm feeling but I'm sure I'm not the only guy who feels this. When a finish a course I'm like ok what can I do right now, how can I test my skills, where and how to start a project, what it's feel like to be in a team, how can I compete with each others...

I hope that make sense.

**Comment by ericdouglas** [comment]
@bewwys I totally agree with you and I know this feeling very well. Thanks a lot for telling us about it, sincerely this is the main issue I see in general about education. 

I'll do my best to think of a solution for this.

**Comment by itsmemattchung** [comment]
> When a finish a course I'm like ok what can I do right now, how can I test my skills, where and how to start a project, what it's feel like to be in a team, how can I compete with each others...

A year and a half ago, shortly after I started working at Amazon in Seattle, I started down the path of studying computer science.  I began researching, asking my colleagues on what courses they suggested as a starting point.  By the time I stumbled upon the OSSU curriculum, I was already three weeks into self-studying a course offered by Carnegie Mellon: Introduction to Computer Systems.  I followed the syllabus published on their website[1], reading from an overly priced textbook[2], streaming their high-quality (recorded) video lectures[3] of the authors themselves.

I loved it.

I acquired a ton of knowledge (some archaic but still intellectually stimulating): calculating floating point numbers and reading assembly and much more. But my favorite topic was an entire chapter (or two) and several recorded lectures devoted to reverse engineering x86 — translating C code to assembly (and vice versa).  To top it off, the authors publish free labs on the book's website[4], challenging students to defuse a binary bomb by running the program through gdb, a debugger.  

But despite all the knowledge I acquired, I felt an important ingredient was missing.  For me, it was not only the _community_ , where I could exchange messages with other students or professors, but most importantly (to me), **feedback**.  Throughout my studies, questions often sprung to mind as I read the textbook or watched the videos; although I was able to sometimes find the answers by searching Google and surfing through various websites and forums, I missed having someone that I could direct my questions to.

Fast forward to today.

I actually signed up for courses from university.  I'm about 2.5 months into taking two distant learning courses—Discrete Mathematics and Computer Organization & Architecture—from two different universities: University of North Dakota and University Northern Iowa. 

I paid somewhere in the range of $800.00 for each course and was initially skeptical of the value that the courses would offer. But contrary to my earlier beliefs, I'm very satisfied with the courses overall.  Similar to OSSU, the discrete mathematics course couples a free textbook with links to supplemental YouTube videos which I find completely sufficient. Another reason reason why I'm pleased, despite paying for the courses, is this: I'm able to contact (i.e. e-mail) the professors and they typically answer my questions within an hour or two (maximum I've seen is 24 hours).  Any time I'm blocked—unclear homework assignment (another great way to reinforce learning) or ambiguous explanation in the videos or text—I'm able to reach out and receive guidance.

These dedicated resources (i.e. professors), in my opinion, make a world of difference.  Not sure how resources can be integrated into OSSU, since the entire idea of OSSU is free and not too many people can afford to serve as dedicated resources to students.  Perhaps people can be allocated for a nominal fee?  I know that goes against the completely *FREE* model but I'm just trying to brain storm some ideas.

[1] Syllabus offered here: http://www.cs.cmu.edu/~213/
[2] $150.00 US for the textbook: https://www.amazon.com/Computer-Systems-Programmers-Perspective-3rd/dp/013409266X
[3] https://scs.hosted.panopto.com/Panopto/Pages/Sessions/List.aspx#folderID=%22b96d90ae-9871-4fae-91e2-b1627b43e25e%22&sortColumn=0&sortAscending=false
[4] http://csapp.cs.cmu.edu/3e/labs.html

**Comment by ericdouglas** [comment]
@itsmemattchung very interesting history, thanks a lot for sharing it with us.

I totally agree with you that feedback is decisive to our success while learning something new, and I have some ideas to provide such thing for OSSU's students.

I'm currently working in parallel on OSSU and in a "meta-project" that aims to give a north for people that want to create a sustainable open source project.

OSSU will implement this framework so we will become a sustainable non-profit organization. One thing will never change though, free educational resources.

With the resources we will gather from different sources, we will pay:

- Course maintainers to create the curriculum and at least one real project for each course of the curriculum, addressing the point @bewwys said before
- Course maintainers can also help providing feedback for students
- Students that want to help others will also be able to receive for it - course assistants

If you have more ideas/suggestions please share with us! :smile:

We'll have a landing page soon so the main features of the OSSU will be there and become more clear :)

**Comment by ericdouglas** [comment]
Some [suggestions](https://github.com/ossu/computer-science/issues/424#issuecomment-333392783) by @alwongg
